lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
   @spec gcd(integer, integer) :: pos_integer
+  @spec gcd(int, int) :: int when int: pos_integer
+  @spec gcd(0, 0) :: 0
```

Comment:
@fxn I based the claim that it cannot be zero on [this proof](https://proofwiki.org/wiki/Existence_of_Greatest_Common_Divisor). Doing some more research, I found [this post on Math StackExchange](http://math.stackexchange.com/a/1387924/20282) which explains what is going on here. Basically it depends on the way you define what is a 'divisor'.

So: Yes, it seems that `gcd(0,0) = 0` is a proper approach. Thank you for your work!